lib/pages/add_bridge/add_bridge.dart

```
@@ -176,6 +190,118 @@ class BotController extends State<AddBridge> {
+  }
+
+  Future<void> pingBridgeAPI(SocialNetwork network) async {
+    final bridgePathIdentifier  = getBridgePath(network);
+    final userId = client.userID;
+
+    final url = 'https://matrix.$serverUrl/_matrix/$bridgePathIdentifier/provision/v3/whoami?user_id=$userId';
+
+    final response = await requestGet(url);
+
+    interpretBridgeResponse(response);
+
+    if (response.statusCode == 200) {
+      if (kDebugMode) {
+        print('Successful ping');
+      }
+    } else {
+      if (kDebugMode) {
+        print('Error: ${response.statusCode}');
+      }
+    }
+  }
+
+  void interpretBridgeResponse(http.Response response) {
+    final responseJson = jsonDecode(response.body);
+
+    final networkName = responseJson['network']?['displayname'];
+    if (networkName != null) {
+      final SocialNetwork? network = SocialNetworkManager.fromName(networkName);
+
+      if (network != null) {
+        final logins = responseJson['logins'];
+        if (logins != null && logins.isNotEmpty) {
+          final stateEvent = logins[0]['state']?['state_event'];
+
+          if (stateEvent == 'CONNECTED') {
+            _updateNetworkStatus(network, true, false);
+            if (kDebugMode) {
+              print('Status: Connected to $networkName');
+            }
+          } else {
+            _updateNetworkStatus(network, false, false);
+            if (kDebugMode) {
+              print('Status: Not connected to $networkName');
+            }
+          }
+        } else {
+          // Treat an empty list as unconnected
+          _updateNetworkStatus(network, false, false);
+          if (kDebugMode) {
+            print('Status: Not connected to $networkName (logins list is empty)');
+          }
+        }
+      } else {
+        if (kDebugMode) {
+          print('Network not detected or not supported: $networkName');
+        }
+      }
+    } else {
+      if (kDebugMode) {
+        print('Network not detected in response.');
```

These errors are not propagated or shown to user